client/components/course/Sidebar/Relationship.vue

```diff
@@ -41,28 +44,45 @@ export default {
     allowEmpty: { type: Boolean, default: true },
     placeholder: { type: String, default: 'Click to select' },
     allowCircularLinks: { type: Boolean, default: false },
-    allowInsideLineage: { type: Boolean, default: false }
+    allowInsideLineage: { type: Boolean, default: false },
+    allowedTypes: { type: Array, default: null }
   },
   computed: {
     ...mapGetters(['activity', 'activities'], 'course'),
     ...mapGetters(['getLineage'], 'activities'),
     options() {
       const { allowInsideLineage, allowCircularLinks, activity: { id } } = this;
-      const activities = without(this.activities, this.activity);
+      let activities = without(this.activities, this.activity);
       const conds = [it => getLevel(it.type)];
       if (!allowCircularLinks) conds.push(it => !includes(this.getAssociationIds(it), id));
       if (!allowInsideLineage) {
         const lineage = this.getLineage(this.activity);
         conds.push(it => !includes(lineage, it));
       }
+      if (this.allowedTypes.length) {
+        conds.push(({ type }) => {
+          const { typeName } = parseType(type);
+          return includes(this.allowedTypes, typeName);
+        });
+      }
       return filter(activities, it => every(conds, cond => cond(it)));
     },
+    optionsGrouped() {
+      return map(groupBy(this.options, 'type'), (it, type) => ({
+        type: parseType(type).typeName,
+        activities: it
+      }));
+    },
     selectPlaceholder() {
       return isEmpty(this.options) ? 'No activities' : this.placeholder;
     },
     associations() {
       const ids = this.getAssociationIds(this.activity);
-      return filter(this.options, it => includes(ids, it.id));
+      const associations = filter(this.options, it => includes(ids, it.id));
+      return this.multiple ? associations : associations[0];
```

Comment:
So return type of this becomes union of `Array(id)` and `id`. Make it always return array (with single member if needed) and pick first inside template depending on `multiple` prop state.